modules/nextflow/src/main/groovy/nextflow/k8s/client/K8sClient.groovy

```diff
@@ -500,6 +503,12 @@ class K8sClient {
             def msg = "K8s pod '$podName' execution failed"
             if( status.reason ) msg += " - reason: ${status.reason}"
             if( status.message ) msg += " - message: ${status.message}"
+            if ( 'OutOfcpu' == status.reason && (status.message as String).startsWith("Pod Node didn't have enough resource: cpu,")) {
+                throw new K8sOutOfCPUException(msg)
+            }
+            if ( 'OutOfmemory' == status.reason && (status.message as String).startsWith("Pod Node didn't have enough resource: memory,") ) {
+                throw new K8sOutOfMemoryException(msg)
+            }
             final err = status.reason == 'Shutdown'
                     ? new NodeTerminationException(msg)
                     : new ProcessFailedException(msg)
```

Comment:
Why don't we use a match expression to handle all of these cases at once? I guess in Groovy 3 you still need to do a regular switch statement, or use a map.

Reply:
Done